Platformer/Assets/Scripts/NPC/Behaviour/MeleeEntityBehaviour.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using UnityEngine;
+
+using Movement.Controller;
+using Movement.Enums;
+
+namespace NPC.Behaviour
+{
+    public class MeleeEntityBehaviour : BaseEntityBehaviour
+    {
+        [SerializeField] private float _attackSpeed;
+        [SerializeField] private Collider2D _entityCollider;

```

Comment:
do the same as the PlayerEntityBehaviour class, extract the collider using Component